frontend/src/components/Form.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { user } from '../reducers/user'
+
+const SIGNUP_URL = 'https://project-auth-lili-destiny-1.herokuapp.com/users'
+const LOGIN_URL = 'https://project-auth-lili-destiny-1.herokuapp.com/sessions'
+
+export const Form = () => {
+  const dispatch = useDispatch()
+
+  const accessToken = useSelector((store) => store.user.login.accessToken)
+
+  const [name, setName] = useState('')
+  const [password, setPassword] = useState('')
+
+  const handleLoginSuccess = (loginResponse) => {
+    dispatch(
+      user.actions.setAccessToken({ accessToken: loginResponse.accessToken })
+    )
+    dispatch(user.actions.setUserId({ userId: loginResponse.userId }))
+    dispatch(user.actions.setStatusMessage({ statusMessage: 'Login Success' })
+    )
+    setName('')
+    setPassword('')
+  }
+
+  const handleLoginFailed = (loginError) => {
+    dispatch(user.actions.setAccessToken({ accessToken: null }))
+    dispatch(user.actions.setStatusMessage({ statusMessage: loginError }))
+  }
+
+  const handleSignup = (event) => {
+    event.preventDefault()
+    
+    fetch(SIGNUP_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, password }),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Signup Failed')
+        }
+        return response.json()
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFailed(err))
+  }
+
+  const handleLogin = (event) => {
+    event.preventDefault()
+
+    fetch(LOGIN_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, password }),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error('Login Failed')
+        }
+        return response.json()
+      })
+      .then((json) => handleLoginSuccess(json))
+      .catch((err) => handleLoginFailed(err))
+  }
+
+  const handleLogout = (loggingout) => { 
+    dispatch(user.actions.logout({ logout: loggingout }))
```

Comment:
Correct user.actions.setLogout instead of user.actions.logout (the reducer is called setLogout)
Also, I don't think it's necessary to send in this object { logout: loggingout } since it is not being used by the reducer at all.
You can probably just write: `dispatch(user.actions.setLogout())`